src/features/User/Ui/NavBar/NavBar.tsx

```diff
@@ -34,11 +35,16 @@ const NavBar = () => {
     setAnchorEl(null);
   };
   const navigate = useNavigate();
-  function logOut(): void {
+  function logOut(){
     localStorage.removeItem("userToken");
```

Comment:
Here we just remove the `userToken` from the local storage, but there is nothing to tell react that the UI has changed. To achieve this, we need to have a state variable holding the userRole, so that when we change it , the UI updates based on this change.